repos/system_upgrade/el7toel8/actors/rpmscanner/libraries/library.py

```diff
@@ -1,11 +1,25 @@
-from leapp.exceptions import StopActorExecutionError
+import warnings
+
+from leapp.exceptions import StopActorExecutionError, UsageError
+
+no_yum = False
+no_yum_warning_msg = "package `yum` is unavailable"
+try:
+    import yum
+except ImportError:
+    no_yum = True
+    warnings.warn(no_yum_warning_msg, ImportWarning)
 
 
 def get_package_repository_data():
-    """ Return dictionary mapping package name with repository from which it was installed """
-    # import has to be inside the function to avoid troubles with non-existing
-    # module in Python3 (where we do not need this function anymore)
-    import yum  # pylint: disable=import-outside-toplevel
+    """ Return dictionary mapping package name with repository from which it was installed.
+    Note:
+        There's no yum module for py3. The dnf module could have been used
+        instead but there's abug in dnf preventing us to do so:
+        https://bugzilla.redhat.com/show_bug.cgi?id=1789840
+    """
+    if no_yum:
+        raise UsageError(message=no_yum_warning_msg)
```

Comment:
I guess this is not exception you want to use here. I think it's purely for the Leapp itself regarding problem with 'how leapp is executed from the commandline'. So I am not sure we should use it in this way. Maybe good time to document Leapp exceptions finally. In this case, use `StopActorExecutionError`.